src/modules/navigator/mission.cpp

```diff
@@ -953,10 +951,10 @@ Mission::altitude_sp_foh_update()
 		* The setpoint is set linearly and such that the system reaches the current altitude at the acceptance
 		* radius around the current waypoint
 		**/
-		float delta_alt = (get_absolute_altitude_for_item(_mission_item) - _mission_item_previous_alt);
+		float delta_alt = (get_absolute_altitude_for_item(_mission_item) - pos_sp_triplet->previous.alt);
```

Comment:
Thought about that too, probably what happens already: no FOH until the first waypoint. So that's something we'd need to add in addition.